rebalance/rebalance.py

```diff
@@ -174,7 +175,7 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi=None,
         for channel in mychannels:
             excludes += [channel['short_channel_id'] + '/0', channel['short_channel_id'] + '/1']
 
-        while int(time.time()) - start_ts < retry_for:
+        while int(time.time()) - start_ts < retry_for and not plugin.rebalance_stop:
```

Comment:
Great idea.
I tried to stop this also for a normal `rebalance` operation (1min default timeout). This does not work to stop it, not sure why, maybe because it does not detach and the `rebalancestop` operation is actually not starting before this finishes.
Can we make this work also for the normal 'rebalance` call?

Reply:
Imho in theory we can have a CLI blocking `rebalance` call and still interrupt this with `rebalancestop`, but I'm not sure if we need changes to `pyln` or even clnd. JSON-RPC is capable of having messages detached from one another. But we maybe need a CLI RPC dispatcher or something so one thread reads the RPC sock ans `fork`'s new requests or notifies them on result.

Reply:
Yep, synchronous plugin methods have a blocking mutex by default, I think. `rebalancestop` will not actually start until normal `rebalance` finishes.

It may a good idea to make `rebalance` non-blocking, but I think it would be too much for now: I'd rather do it later in a different PR. It raises some questions also: if `rebalance` is non-blocking, then we cannot easily see when it finishes and what is its return value.

Reply:
Yep, your right.